lib/active_model/serializer/fragment_cache.rb

```diff
@@ -0,0 +1,99 @@
+        else
+          non_cached_hash.merge cached_hash
+        end
+      end
+
+      private
+
+      def cached_hash_root(cached_hash, non_cached_hash)
+        hash = {}
+
+        core_cached = cached_hash.first
+        core_non_cached = non_cached_hash.first
+        no_root_cache = cached_hash.delete_if {|key, value| key == core_cached[0] }
+        no_root_non_cache = non_cached_hash.delete_if {|key, value| key == core_non_cached[0] }
+
+        if @root
+          hash[@root] = (core_cached[1]) ? core_cached[1].merge(core_non_cached[1]) : core_non_cached[1]
+        else
+          hash = (core_cached[1]) ? core_cached[1].merge(core_non_cached[1]) : core_non_cached[1]
+        end
+        hash.merge no_root_non_cache.merge no_root_cache
+      end
+
+      def fragment_associations(serializers, associations)
+        associations.each do |association|
+          options = ",#{association[1][:association_options]}" if association[1].include?(:association_options)
+          eval("#{serializers[:non_cached]}.#{association[1][:type].to_s}(:#{association[0]}#{options})")
+        end
+      end
+
+      def cached_attributes_and_association(klass, serializers)
+        cached_attr = (klass._cache_only) ? klass._cache_only : @serializer.attributes.keys.delete_if {|attr| klass._cache_except.include?(attr) }
+        non_cached_attr = @serializer.attributes.keys.delete_if {|attr| cached_attr.include?(attr) }
+        associations = @serializer.each_association
+
+        cached_attr.each do |attr|
+          if @serializer.each_association.keys.include?(attr)
+            associations.delete(attr)
+            serializers[:cached].constantize.send(@serializer.each_association[attr][:type], attr)
+          end
+        end
+
+        cached_attr.each do |attribute|
+          options = @serializer.class._attributes_keys[attribute]
+          options ||= {}
+          serializers[:cached].constantize.attribute(attribute, options)
+        end
+
+        non_cached_attr.each do |attribute|
+          options = @serializer.class._attributes_keys[attribute]
+          options ||= {}
+          serializers[:non_cached].constantize.attribute(attribute, options)
+        end
+        fragment_associations(serializers, associations)
+      end
+
+      def fragment_serializer(name, klass)
+        cached = "#{name.capitalize}CachedSerializer"
+        non_cached = "#{name.capitalize}NonCachedSerializer"
+
+        Object.const_set cached, Class.new(ActiveModel::Serializer) unless Object.const_defined?(cached)
+        Object.const_set non_cached, Class.new(ActiveModel::Serializer) unless Object.const_defined?(non_cached)
+
+        klass._cache_options ||= {}
+        klass._cache_options[:key] = klass._cache_key if klass._cache_key
+        cached.constantize.cache(klass._cache_options)
+
+        serializers = {cached: cached, non_cached: non_cached}
+        cached_attributes_and_association(klass, serializers)
+        return serializers
```

Comment:
Yeah, I know, I just felt that I should make it explicitly.
Never mind I'm removing it. :stuck_out_tongue_closed_eyes: